fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java

```diff
@@ -2223,6 +2225,9 @@ public Optional<Boolean> isFollowerForwardToLeaderOpt() {
     @VarAttr(name = LIKE_PREDICATE_CONSOLIDATE_MIN)
     private int likePredicateConsolidateMin = 2;
 
+    @VarAttr(name = CUSTOM_QUERY_ID)
+    private String customQueryId = "";
```

Comment:
As you mentioned in the [#47699](https://github.com/StarRocks/starrocks/issues/47699), user would ensure the unique property, but should StarRocks add a check to tell user you are setting a duplicate  custom query id?

Reply:
It's time comsuming to ensure `custom_query_id` globally unique accross all FE(leader, follower, observer). And if one FE is hung, this check would be hung too. So I don't think it's a good choice to do this check.

Usually, only expert user will set this `custom_query_id`, like a platform user who can ensure the `custom_query_id` is unique in his program.